### DISCUSSION TITLE
SEB Reset Utility failed to reset 8 items

### DISCUSSION THREAD

**Discussion by oliahim98** [SafeExamBrowser/seb-win-refactoring#943] [Q&A]
I have a problem when running the SEB Reset Utility, I am selecting the option to reset system configuration to default values, and this is the result. I attached the log file bellow. I am using the SEB version 3.5.0 on Windows 10.
![image](https://github.com/user-attachments/assets/ab767cda-1be2-4924-8c06-3973768d7b35)
[2024-07-30_13h10m33s_ResetUtility.log](https://github.com/user-attachments/files/16426714/2024-07-30_13h10m33s_ResetUtility.log)

**Comment by dbuechel** [comment]
Thanks for the report, we'll have a look at the issue.

**Comment by dbuechel** [comment]
The issue appears to happen when attempting to delete all user-related items, e.g.:

```
2024-07-30 13:10:48.171 [01] - ERROR: [ChangePasswordConfiguration] Failed to delete registry item 'HKEY_USERS\S-[...]-1001\Software\Microsoft\Windows\CurrentVersion\Policies\System\DisableChangePassword'!

   Exception Message: Object reference not set to an instance of an object.
   Exception Type: System.NullReferenceException

   at SafeExamBrowser.Lockdown.FeatureConfigurations.RegistryConfigurations.RegistryConfiguration.TryDelete(RegistryDataItem item) in C:\Users\appveyor\projects\seb-win-refactoring-ist041kp6jai2xxx\SafeExamBrowser.Lockdown\FeatureConfigurations\RegistryConfigurations\RegistryConfiguration.cs:line 192
```

It is at the moment unclear to me how that could happen, since we're checking the availability of the respective registry hive before accessing it. Did you make sure that you were logged-in with the exact same user under which you ran SEB previously (resp. for which you want to reset the configuration)? Which exact operating system version and edition are you using (e.g. Windows 10 Pro 22H2)?

Also, please open the registry editor (search the Windows start menu or type `regedit.exe`), paste the following path into the address bar and provide us a screen shot of the result: `HKEY_CURRENT_USER\Software\Microsoft\Windows\CurrentVersion\Policies\System\`.

**Reply by oliahim98** [reply]
I am running the tool on a personal computer. There is only one user, so that is not the problem. I am on Windows 10 Pro 22H2.
The registry entries might be where the problem lies. There is no System folder at all here: HKEY_CURRENT_USER\Software\Microsoft\Windows\CurrentVersion\Policies
![image](https://github.com/user-attachments/assets/c65dee0a-3794-4123-8890-2b034488e16a)

**Reply by dbuechel** [reply]
Thanks for the information, that might then indeed be an issue. I shall investigate this further and report back once I know more.

**Reply by dbuechel** [reply]
> The registry entries might be where the problem lies. There is no System folder at all here: `HKEY_CURRENT_USER\Software\Microsoft\Windows\CurrentVersion\Policies`

That is indeed the cause of the issue (there are two subfolders, which SEB currently expects to be there). We'll fix this in an upcoming version, but for the time being you can simply ignore these errors (as they only indicate that deleting something which doesn't exist has failed).

**Reply by dbuechel** [reply]
We'll track this under #956.
